src/cart/crt_iv.c

```diff
@@ -1346,37 +1359,42 @@ crt_hdlr_iv_fetch_aux(void *arg)
 			"Group (%s) version mismatch. Local: %d Remote :%d\n",
 			ivns_id.ii_group_name, grp_ver_entry,
 			input->ifi_grp_ver);
-		D_GOTO(send_error, rc = -DER_GRPVER);
+		D_GOTO(reply_direct, rc = -DER_GRPVER);
 	}
 
 	iv_ops = crt_iv_ops_get(ivns_internal, input->ifi_class_id);
 	if (iv_ops == NULL) {
 		D_ERROR("Returned iv_ops were NULL\n");
-		D_GOTO(send_error, rc = -DER_INVAL);
+		D_GOTO(reply_direct, rc = -DER_INVAL);
 	}
 
 	IV_DBG(&input->ifi_key, "fetch handler entered\n");
 	rc = iv_ops->ivo_on_get(ivns_internal, &input->ifi_key,
 				0, CRT_IV_PERM_READ, &iv_value, &user_priv);
 	if (rc != 0) {
 		D_ERROR("ivo_on_get(): "DF_RC"\n", DP_RC(rc));
-		D_GOTO(send_error, rc);
+		D_GOTO(reply_direct, rc);
 	}
 
 	put_needed = true;
-
 	rc = iv_ops->ivo_on_fetch(ivns_internal, &input->ifi_key, 0,
 				  0x0, &iv_value, user_priv);
 	if (rc == 0) {
 		/* Note: This increments ref count on 'rpc_req' and ivns */
-		rc = crt_ivf_bulk_transfer(ivns_internal,
-					   input->ifi_class_id,
-					   &input->ifi_key,
-					   &iv_value, input->ifi_value_bulk,
-					   rpc_req, user_priv);
-		if (rc != 0) {
-			D_ERROR("bulk transfer failed; "DF_RC"\n", DP_RC(rc));
-			D_GOTO(send_error, rc);
+		if (input->ifi_value_bulk != CRT_BULK_NULL) {
+			rc = crt_ivf_bulk_transfer(ivns_internal,
+						   input->ifi_class_id,
+						   &input->ifi_key,
+						   &iv_value, input->ifi_value_bulk,
+						   rpc_req, user_priv);
+			if (rc != 0) {
+				D_ERROR("bulk transfer failed; "DF_RC"\n", DP_RC(rc));
```

Comment:
we should start using DL_ERROR() macro variations when rc is present for the new code

Reply:
sure